### ISSUE TITLE
[Feat] Frozen backbone finetuning

### ISSUE THREAD

**Issue by Jaster1999** [roboflow/rf-detr#783]
### Search before asking

- [x] I have searched the RF-DETR [issues](https://github.com/roboflow/rf-detr/issues) and found no similar feature requests.

### Description

The ability to set the backbone to be frozen for finetuning. something like freeze_backbone=True in model.train args e.g.

model.train(
   dataset_dir="",
   freeze_backbone=True,
)

### Use case

Specifically this would be useful for training very large models on consumer gpu hardware but also to prevent overfitting on smaller datasets. The logic here being the object365 pretraining can create generalized features for object detection and we just adapt the object detection neck and head for the downstream task.

If it works we could potentially train more generalized models on small dataset (sacrificing better metrics for generalization)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by isaacrob** [comment]
we decay the learning rate through the backbone already, you should be able to make this happen by setting lr_vit_layer_decay=0

**Comment by Jaster1999** [comment]
Ok cool, how does that rate work exactly as in it will multiply by the lr e.g. layer lr = lr_vit_layer_decay * lr?

**Comment by Jaster1999** [comment]
Thinking about this further, setting lr=0 wouldn't disable gradients for the backbone no? Ideally if we were effectively disabling the gradients via lr=0, we'd want the backbone to just not have gradients as it would be much faster to train and use less vram (e.g. if backbone was frozen in eval mode or disconnected from backprop).

**Comment by isaacrob** [comment]
True! Yeah wouldn't have those benefits. Prob worth adding feature then. Want to write pr?

**Comment by zhaoshuo1223** [comment]
> 没错！那样就没那些好处了。看来值得添加这个功能。想写个 PR 吗？

In main.py, I noticed that there seems to be a freeze_encoderparameter that will freeze DINOV2.

**Comment by VijayVignesh1** [comment]
I am not able to see the freeze_backbone option. I can raise a PR to add this feature.

**Comment by Borda** [comment]
🎉 Implemented in v1.6.0! Use the `freeze_encoder` parameter in `ModelConfig`:
```python
from rfdetr import RFDETRBase
model = RFDETRBase(freeze_encoder=True)
model.train(dataset="your_dataset", ...)
```

To upgrade: `pip install rfdetr>=1.6.1`

Closing as implemented. Feel free to reopen if the behavior doesn't match expectations!
